swarm/api/routes/db.py

```
@@ -284,6 +297,14 @@ async def ingest_run_events(run_id: str):
         Dict with ingestion statistics.
     """
     try:
+        try:
+            validate_path_component(run_id, "run_id")
+        except ValueError as e:
+            raise HTTPException(
+                status_code=400,
+                detail={"error": "invalid_request", "message": str(e)},
+            )
```

Ruff (0.14.14)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)